tests/rest_scripts/rest.sh

```diff
@@ -65,3 +65,18 @@ $canonical_request_hash"
   curl_command=()
   add_command_recording_if_enabled
 }
+
+add_parameter() {
+  if [ "$#" -ne 2 ]; then
+    return
+  fi
+  if [ "$first_param_added" != "true" ]; then
+    if [ "$1" == "url" ]; then
+      eval "$1"+='?'
+    fi
+    eval "$1"+="$2"
+    first_param_added="true"
```

Comment:
Having this global var controlled outside the function seems really risky to me - it assumes a 'correct' way to use this and could generate weird bugs when you don't reset it.  Maybe you could parse the url to see what state it is in to correctly add first/additional param?

Reply:
Yeah, I'll try to fix this